src/transformers/models/kosmos2_5/modeling_kosmos2_5.py

```diff
@@ -0,0 +1,2403 @@
+            if module.key.bias is not None:
+                module.key.bias.data.zero_()
+            if module.value.bias is not None:
+                module.value.bias.data.zero_()
+            if module.output.bias is not None:
+                module.output.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5VisionMlp):
+            in_proj_std = (module.config.hidden_size**-0.5) * ((2 * module.config.num_hidden_layers) ** -0.5) * factor
+            fc_std = (2 * module.config.hidden_size) ** -0.5 * factor
+            nn.init.normal_(module.wi_0.weight, std=fc_std)
+            nn.init.normal_(module.wi_1.weight, std=in_proj_std)
+            nn.init.normal_(module.wo.weight, std=fc_std)
+            if module.wi_0.bias is not None:
+                module.wi_0.bias.data.zero_()
+            if module.wi_1.bias is not None:
+                module.wi_1.bias.data.zero_()
+            if module.wo.bias is not None:
+                module.wo.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5VisionLayer):
+            module.pre_mlp_layer_norm.weight.data.fill_(1.0)
+            module.pre_attention_layer_norm.weight.data.fill_(1.0)
+        elif isinstance(module, Kosmos2_5TextAttention):
+            nn.init.normal_(module.q_proj.weight, std=std)
+            nn.init.normal_(module.k_proj.weight, std=std)
+            nn.init.normal_(module.v_proj.weight, std=std)
+            nn.init.normal_(module.out_proj.weight, std=std)
+            if module.q_proj.bias is not None:
+                module.q_proj.bias.data.zero_()
+            if module.k_proj.bias is not None:
+                module.k_proj.bias.data.zero_()
+            if module.v_proj.bias is not None:
+                module.v_proj.bias.data.zero_()
+            if module.out_proj.bias is not None:
+                module.out_proj.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5TextFFN):
+            nn.init.normal_(module.fc1.weight, std=std)
+            nn.init.normal_(module.fc2.weight, std=std)
+            if module.fc1.bias is not None:
+                module.fc1.bias.data.zero_()
+            if module.fc2.bias is not None:
+                module.fc2.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5TextForCausalLM):
+            nn.init.normal_(module.lm_head.weight, std=std)
+            if module.lm_head.bias is not None:
+                module.lm_head.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5ImageToTextProjection):
+            nn.init.normal_(module.dense.weight, std=std)
+            if module.dense.bias is not None:
+                module.dense.bias.data.zero_()
+        elif isinstance(module, Kosmos2_5TextTransformer):
+            module.embed_tokens.weight.data.normal_(std=std)
+            if module.embed_tokens.padding_idx is not None:
+                module.embed_tokens.weight.data[module.embed_tokens.padding_idx].zero_()
+            module.segment_emb.weight.data.normal_(std=std)
```

Comment:
let's simplify this as zero is used for any / almost all bias, and std is the same for a lot of modules let's have smart defualts

Reply:
After I remove bias.data.zero_()
```
Kosmos2_5ModelTest::test_initialization - AssertionError: 0.00976146012544632 not found in [0.0, 1.0] : Parameter text_model.model.layers.0.self_attn.k_proj.bias of model <class 'transformers.models.kosmos2_5.modeling_kosmos2_5.Kosmos2_5Model'> seems not properly initialized
```

Reply:
I don't think he means `remove bias.data.zero_()`. Let me check with him.

Reply:
done